api/src/events.py

```diff
@@ -0,0 +1,625 @@
+    user = db.users.find_one({"_id": ObjectId(user_id)})
+    if user == None:
+        return Response('Invalid request', status=400)
+    registrations = user['registrations']
+
+    return jsonify(registrations)
+
+
+@events_api.route('/getParticipants/<event>', methods=['GET'])
+@jwt_required
+def get_participant(event):
+
+    """Get all participants for a specific event.
+
+    :param event_name: event name
+
+    :reqjson hopkins_student: Used if and only if ONLY Hopkins students should be returned
+    :reqjson reg_status: registration status of users being returned
+
+    Example input:
+
+    .. sourcecode:: json
+
+    {
+        "hopkins_student": "ye",
+        "reg_status": "rsvp"
+    }
+
+    Both hopkins_student and reg_status are essentially search filters
+
+    :return: dictionary with 'participants' as key and value as list of all participants of the event
+
+    :status 200: list of all participants is returned successfully
+    :status 400: Invalid request or event does not exist
+
+    """
+
+    if event is None:
+        return Response('Invalid request', status=400)
+
+    args = {'event_name': event}
+    if len(list(db.events.find(args))) == 0:
+        return Response('Event Does Not Exist', status=400)
+
+    # Do not use hopkins_student in arguments (request.json) unless you only want Hopkins Students
+    # If a user being a Hopkins student is irrelevant, just do not include "hopkins_student" in request.json
+    participants = []
+    cursor = db.events.find(args)
+    if 'hopkins_student' in request.args and 'reg_status' in request.args:
+        for i in cursor:
+            for j in i['event_participants']:
+                if j['status'] == request.args['reg_status'] and j['profile']['is_jhu'] == True:
+                    participants.append(j)
+            break
+    elif 'hopkins_student' in request.args:
+        for i in cursor:
+            for j in i['event_participants']:
+                print(j['profile']['school'])
+                if j['profile']['is_jhu'] == True:
+                    participants.append(j)
+            break
+    elif 'reg_status' in request.args:
+        for i in cursor:
+            for j in i['event_participants']:
+                if j['status'] == request.args['reg_status']:
+                    participants.append(j)
+            break
+    else:
+        for i in cursor:
+            participants = i['event_participants']
+            break
+    return jsonify({'participants': participants}), 200
+
+
+def remove_duplicates_from_list(random_list):
+    """Removes duplicates from a list
+
+    :param random_list: List that duplicates will be removed from
+    :return: list without any duplicate values
+
+    """
+    res = []
+    user_ids = set()
+    for i in random_list:
+        if i["user_id"] not in user_ids:
+            res.append({"profile": i["profile"], "username": i["username"]})
+            user_ids.add(i["user_id"])
+    return res
```

Comment:
Python Hack
res = []
[res.append(x) for x in test_list if x not in res]

Reply:
Realized that [res.append(x) for x in test_list if x not in res] wouldn't work if the same user registered for two events but had two different registration status for those two different events. So, modified it to [res.append({"profile": x["profile"], "username": x["username"]}) for x in random_list if {"profile": x["profile"], "username": x["username"]} not in res]